src/main/java/seedu/storemando/model/item/ExpiryDate.java

```diff
@@ -37,19 +40,28 @@
     public ExpiryDate(String expiryDate) {
         requireNonNull(expiryDate);
         checkArgument(isValidExpiryDate(expiryDate), MESSAGE_CONSTRAINTS);
-        value = expiryDate;
+        value = LocalDate.parse(expiryDate);
     }
 
     /**
      * Returns if a given string is a valid expiryDate.
      */
     public static boolean isValidExpiryDate(String test) {
-        return test.matches(VALIDATION_REGEX);
+        requireNonNull(test);
+        if (test == null || !test.matches(VALIDATION_REGEX)) {
```

Comment:
Just a suggestion. Would it be better if you put everything in a try-catch block?
try {
    if (test.matches(VALIDATION_REGEX)) {
        LocalDate.parse(test);
        return true;
    }
} catch (DateTimeParseException | NullPointerException ex) {
    return false;
}

Reply:
Thanks for the suggestion. Fixed.